Enable Travis-CI

ctk3b opened the pull request.

ctk3b committed:
Enable Travis-CI

ctk3b committed:
Merge remote-tracking branch 'upstream/master' into travis-ci

csadorf committed:
Fix symbolic links.

csadorf committed:
Merge branch 'develop' into travis-ci

csadorf committed:
Deactivate Python2.7 testing.

Both signac-flow and the template are currently not Python2.7
compatible.

ctk3b committed:
Try different osx images

ctk3b committed:
Merge remote-tracking branch 'upstream/travis-ci' into travis-ci

csadorf committed:
Remove sysctl command.

csadorf committed:
Test xcode8.1

csadorf commented:
The OS-X tests are failing, because hoomd is trying to execute AVX instructions, which are not available on the testing platform. Trying different xcode versions does not solve the issue.

I'm going to mock the hoomd package, since running the (short), but still slightly expensive demo simulations is not critical for testing the template. That would have the additional benefit that test runs are shorter and cheaper on all platforms.

csadorf committed:
Disable testing on Mac OS-X platform.

Testing on Mac OS-X images currently not possible due to limitations
with executing the conda/glotzer/hoomd package on the os-x images.

It is not expected that the template itself would behave differently on
the os-x platform anyways.
